### PR TITLE
Cp 496 logging package

### PR DESCRIPTION
**Pull request by bogdanbudisan**
added a shared package for writing logs

### CODE REVIEW

File: core/shared/lib/src/logger/abstract_logger.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  void debug(String message);
+
+  void info(String message);
+
+  void warning(String message);
+
+  void error(String message);
+
+  void fatal(String message);
```
Comment (francescoberardi) [inline core/shared/lib/src/logger/abstract_logger.dart]: We should add documentation comment for  (/// .....)

File: core/shared/lib/src/logger/logger_util.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  error,
+  fatal,
+}
+
+// Concrete implementation of AbstractLogger using the Logger library
+// Should the logger library change or be replaced,
+// then only this class needs to change, while its usages would stay the same
+class LoggerUtil extends AbstractLogger {
+
+  var logger = Logger();
```
Comment (francescoberardi) [inline core/shared/lib/src/logger/logger_util.dart]:
```suggestion
  final Logger logger;
  
    LoggerUtil({required this.logger});
```

We should pass the external dependencies in the constructor

Comment (francescoberardi) [reply core/shared/lib/src/logger/logger_util.dart]: And then register the external dependency and the LoggerUtil inside the shared_injection_module
```suggestion
    InjectionModule.injector.registerLazySingleton<AbstractLogger>(
        () => LoggerUtil(logger: InjectionModule.injector()));
    InjectionModule.injector.registerLazySingleton<Logger>(() => Logger());
```

File: features/feature1/lib/src/presentation/pages/feature1_page.dart [hunk 3]
```diff
@@ -28,11 +28,14 @@ class Feature1BodyPage extends StatelessWidget {
   Widget build(BuildContext context) {
     return BlocBuilder<Feature1Bloc, Feature1State>(
       builder: (context, state) {
+        LoggerUtil().info("current state: $state");
+
```
Comment (francescoberardi) [inline features/feature1/lib/src/presentation/pages/feature1_page.dart]: After previous changes here we could access the logger doing something like this:
```suggestion
 InjectionModule.injector<AbstractLogger>().info("current state: $state");

```

File: core/shared/lib/shared.dart [hunk 4]
```diff
@@ -7,3 +7,4 @@ export './src/module/shared_injection_module.dart';
 export './src/network/network_info.dart';
 export './src/usecases/usecase.dart';
 export './src/usecases/usecase_params.dart';
+export './src/logger/logger_util.dart';
```
Comment (francescoberardi) [inline core/shared/lib/shared.dart]: We should export only the abstract_logger.dart to the other packages, hiding the real implementation

**Review by francescoberardi — CHANGES_REQUESTED** [review]
I add some changes proposals.
Let me know if all is clear and If you agree with them.

**Comment by bogdanbudisan** [comment]
I later decided to move the LoggerLevel enum to the abstract class, to have them as part of the abstraction rather than the implementation, and to avoid having the abstract class depend on the implementation class
If a different library is used, the LoggerLevels should all stay the same as well

But I'm not sure whether the LoggerLevels are expected to change in the future or not (?)

**Comment by bogdanbudisan** [comment]
> 
> 
> I add some changes proposals.
> Let me know if all is clear and If you agree with them.

**Review by francescoberardi — APPROVED** [review]

**Pull request merged**